framework/doc/content/source/userobjects/NodalPatchRecoveryVariable.md

```diff
@@ -0,0 +1,11 @@
+# NodalPatchRecoveryVariable
```

Comment:
Thoughts about making this a Corrector? It's another system without a base class which is iffy, but in the future there might be some functionality we can share

Reply:
Yeah, the patch recovery capability is pretty standalone in the sense that it is responsible for projecting some field onto arbitrary global function space, and that task is fairly generic.

It doesn't prevent you from implementing a patch-recovery-based corrector though. But that should probably be a separate class that calls into this one.

Now that we are on this subject, as a related note, I think we would benefit from moving the patch recovery logic into the framework itself (e.g., integrating it into SubProblem or perhaps SystemBase). I'm probably guilty of implementing too many capabilities as userobjects... I'll think more about this, and please let me know what your opinions are.

Reply:
I'm always all for upstreaming things